www/docs/src/components/theme-select/theme-select.tsx

```diff
@@ -1,53 +1,54 @@
-import { useEffect } from 'react'
-import { useStore } from '@/state'
-import type { State } from '@/state/types'
+import { useEffect, useState } from 'react'
 import { SunMedium, Moon, LaptopMinimal } from 'lucide-react'
 import type { LucideIcon } from 'lucide-react'
 import clsx from 'clsx'
 
+type Theme = 'system' | 'light' | 'dark'
+
 interface Control {
   ariaLabel: string
   icon: LucideIcon
-  name: State['theme']
+  name: Theme
 }
 
 const ThemeSwitcher = () => {
-  const { theme, getPreferredColorScheme, updateTheme } = useStore()
-
-  const onThemeChange = (selectedTheme: State['theme']) => {
-    const effectiveTheme =
-      selectedTheme === 'auto' ?
-        getPreferredColorScheme()
-      : selectedTheme
-    document.documentElement.dataset.theme = effectiveTheme
-    updateTheme(selectedTheme)
-  }
+  const [themeState, setThemeState] = useState<Theme>(() => {
+    const theme =
+      (localStorage.getItem('theme') as Theme | null) ?? 'system'
```

Comment:
small nit:
```suggestion
    const theme = (localStorage.getItem('theme') ?? 'system') as Theme
```

same behavior, but i find this pattern cleaner to make both eslint and tsc happy